### PR TITLE
add circuit breaker sample

### PR DESCRIPTION
**Pull request by rosensilva**
## Automation tests
 - Unit tests 
   >  done
 - Integration tests
   > N/A

## Security checks
 - Ran FindSecurityBugs plugin and verified report? no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

**Comment by rosensilva** [comment]
I have made the changes according to the recommendations. Please point out if something is still missing.

File: orderServices/order_service.bal [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+@http:configuration {basePath:"/order"}
+service<http> orderService {
+    // The resilient HTTP endpoint is defined including the  Circuit Breaker pattern
+    endpoint<resiliency:CircuitBreaker> circuitBreakerEP {
+        // The Circuit Breaker should be initialized with HTTP Client, failure threshold and reset timeout
+        // HTTP client could be any HTTP endpoint that have risk of failure
+        // Failure threshold should be 0 and 1
+        // reset timeout for circuit breaker should be in milliseconds
+        create resiliency:CircuitBreaker(create http:HttpClient("http://localhost:9092", null), 0.2, 20000);
```
Comment (kasun04) [inline orderServices/order_service.bal]: Explain the CB parameter in detail.

File: README.md [hunk 2]
```diff
@@ -1 +1,222 @@
...
+ 
+- JDK 1.8 or later
+- Ballerina Distribution (Install Instructions:  https://ballerinalang.org/docs/quick-tour/quick-tour/#install-ballerina)
+- A Text Editor or an IDE 
+
+Optional Requirements
+- Docker (Follow instructions in https://docs.docker.com/engine/installation/)
+- Ballerina IDE plugins. ( Intellij IDEA, VSCode, Atom)
+- Testerina (Refer: https://github.com/ballerinalang/testerina)
+- Container-support (Refer: https://githu{"Error":"Inventory Service did not respond","Error_message":"Upstream service unavailable. Requests to upstream service will be suspended for 14451 milliseconds."}b.com/ballerinalang/container-support)
```
Comment (kasun04) [inline README.md]: Typo or malformed content?

File: README.md [hunk 3]
```diff
@@ -1 +1,222 @@
...
+        http:OutRequest outRequest = {};
+        // Initialize the response message to send back to client
+        http:InResponse inResponse = {};
+        http:HttpConnectorError err;
+        // Extract the items from the json payload
+        json items = request.getJsonPayload().items;
+        // Send bad request message to the client if request don't contain items JSON
+        if (items == null) {
+            outResponse.setStringPayload("Error : Please check the input json payload");
+            outResponse.statusCode = 400;
```
Comment (kasun04) [inline README.md]: Good Job. Also, you can explain the reason for setting 400 (i.e. Bad request).

File: README.md [hunk 4]
```diff
@@ -1 +1,222 @@
-# resiliency-circuit-breaker
\ No newline at end of file
+# Circuit Breaker
+This guide walks you through the process of adding Circuit Breaker pattern to potentially-failing remote backend. Circuit Breaker pattern is a way to automatically degrade functionality when remote services fail. Use of the Circuit Breaker pattern can allow a web service to continue operating without waiting for unresponsive remote services.
```
Comment (kasun04) [inline README.md]: Link CB into https://martinfowler.com/bliki/CircuitBreaker.html.

**Pull request merged**